modules/tensor_mechanics/src/actions/TensorMechanicsActionBase.C

```diff
@@ -140,3 +144,42 @@ TensorMechanicsActionBase::outputPropertiesType()
 
   return MultiMooseEnum(options);
 }
+
+void
+    TensorMechanicsActionBase::addRelationshipManagers(Moose::RelationshipManagerType /*when_type*/)
```

Comment:
Actually, it looks like @permcody implemented this in MooseObjectAction...this seems like a hacky fix, but should work?

Reply:
`addRelationshipManagers()` is called for all Actions, not just `MooseObjectAction` so that method will get called. The only thing I don't like is the code duplication in the `addSpecificRelationshipManagers()` method. 

There will be more of these `Actions` that do need to interact with the RelationshipManager system so we might need to come up with an interface that makes this easy to do (that's precociously why we have an implementation already set up in `MooseObjectAction`), but I understand why you can't do that here. I'm not too worried about it right now as we have a rewrite of the RelationshipManger setup being worked on right now. We need to let the dust settle before we come back and address this Action.

Reply:
The duplication is so I can pass in the params instead of relying on _moose_object_pars. I guess I could have done
```
TensorMechanicsActionBase::addSpecificRelationshipManagers(InputParameters param)
{
  _moose_object_pars = param;
  MooseObjectACtion::add
RelationshipManagers(Moose::RelationshipManagerType /*when_type*/)
}
```
or something like that.

But anyways, it sounds like this may change soon anyways, so can I safely ignore it for now? What are the conditions where this will start to break?

Reply:
nonlocal solution access and/or DistributedMesh - neither of which are being used with TensorMechanics right now